ONSdigital/eq-survey-runner
# Make session timeout configurable via the schema

### What is the context of this PR?
The session timeout was previously made configurable by using session storage, but this won't work when using phantomjs for testing. This change also allows the timeout to be configured per survey if required.

As part of this pull request the grace period has been changed so that it is added to the server side session rather than subtracted from the timeout period sent to the client. This is so that when the timeout is defined in the schema, that is the timeout the client experiences before being timed out. 

### How to review 
- Check out and use the `test_timeout.json` survey. It should show the prompt after 5 seconds and timeout after 10 seconds.
- Run the `timeout.spec.js` test
